chain/transaction.go

```diff
@@ -377,6 +379,81 @@ func (t *Transaction) Marshal(p *codec.Packer) error {
 	return t.marshal(p)
 }
 
+type txJSON struct {
+	ID      ids.ID      `json:"id"`
+	Actions codec.Bytes `json:"actions"`
+	Auth    codec.Bytes `json:"auth"`
+	Base    codec.Bytes `json:"base"`
+}
+
+func (t *Transaction) MarshalJSON() ([]byte, error) {
+	actionsPacker := codec.NewWriter(0, consts.NetworkSizeLimit)
+	err := t.Actions.MarshalInto(actionsPacker)
+	if err != nil {
+		return nil, err
+	}
+	if err := actionsPacker.Err(); err != nil {
+		return nil, err
+	}
+	authPacker := codec.NewWriter(0, consts.NetworkSizeLimit)
+	authPacker.PackByte(t.Auth.GetTypeID())
+	t.Auth.Marshal(authPacker)
+	if err := authPacker.Err(); err != nil {
+		return nil, err
+	}
+	basePacker := codec.NewWriter(0, consts.NetworkSizeLimit)
+	if t.Base != nil {
+		t.Base.Marshal(basePacker)
+	}
```

Comment:
What's the reason for this check? Base should never be nil.

Reply:
it should not, but as it is a pointer I prefered to check it instead of risking a panic 🙂 but anyway we don't need to use the codec as you suggested in another comment 👍